designs/0034-jacobian.md

```diff
@@ -0,0 +1,195 @@
+- Feature Name: `jacobian` target
+- Start Date: 01-25-2024
+
+# Summary
+[summary]: #summary
+
+This design doc proposes adding a `jacobian` target and `*_jacobian` user defined functions. The `jacobian` target will be accessible directly in `transformed parameters` and the new `*_jacobian` functions.
+
+
+The examples below show an example use case.
+
+```stan
+functions {
+  real upper_bound_jacobian(real x, real upper_bound) {
+    jacobian += x;
+    return upper_bound - exp(x);
+  }
+}
+data {
+ real ub;
+}
+parameters {
+ // User defined constraint
+ real b_raw;
+}
+transformed paramters {
+  real b = upper_bound_jacobian(b_raw, ub);
+}
+```
+
+# Motivation
+[motivation]: #motivation
+
+Given a function $c$ mapping unconstrained parameters in $Y$ to constrained parameters in $X$, probability density function $\pi$, and a Jacobian determinant function over the constrained space $J(c(y))$, Stan calculates the log transformed density function [1]
+
+$$
+\pi^*(y) = \pi\left( c\left(y\right) \right) J_c\left(y\right)
+$$
+
+$$
+\log\left(\pi^*(y)\right) = \log\left(\pi\left( c\left(y\right) \right)\right) + \log\left(J_c\left(y\right)\right)
+$$
+
+The Stan languages has built in constraints constraints such as `lower`, `upper`, `ordered`, etc. to handle $\log(J_c(y))$. 
+A variable (unconstraining) transform is a surjective function $f:\mathcal{X} \rightarrow \mathbb{R}^N$ from a constrained subset $\mathcal{X} \subseteq \mathbb{R}^M$ onto the full space $\mathbb{R}^N$.
+The inverse transform $f^{-1}$ maps from the unconstrained space to the constrained space. 
+Let $J$ be the Jacobian of $f^{-1}$ so that $J(x) = (\nabla f^{-1})(x).$ and $|J(x)|$ is its absolute Jacobian determinant. A transform in Stan specifies 
+
+- $f(y)$ The unconstraining transform
+- $f^{-1}\left(y\right)$ The inverse unconstraining transform
```

Comment:
This should be "inverse of the unconstraining transforming" because it's actually constraining.  I'd just call it "the constraining transform".